src/modules/75-cd/components/EnvironmentGroups/EnvironmentGroupDetails/EnvironmentGroupDetails.tsx

```diff
@@ -0,0 +1,435 @@
+  EnvironmentGroupResponseDTO,
+  EnvironmentResponse,
+  updateEnvironmentGroupPromise,
+  useGetEnvironmentGroup,
+  useGetYamlSchema
+} from 'services/cd-ng'
+
+import type {
+  EnvironmentGroupPathProps,
+  EnvironmentGroupQueryParams,
+  ProjectPathProps
+} from '@common/interfaces/RouteInterfaces'
+import { yamlStringify } from '@common/utils/YamlHelperMethods'
+import YAMLBuilder from '@common/components/YAMLBuilder/YamlBuilder'
+import { NameIdDescriptionTags } from '@common/components'
+import { getScopeFromDTO } from '@common/components/EntityReference/EntityReference'
+import type { YamlBuilderHandlerBinding, YamlBuilderProps } from '@common/interfaces/YAMLBuilderProps'
+import { useDeepCompareEffect, useQueryParams, useUpdateQueryParams } from '@common/hooks'
+import { IdentifierSchema, NameSchema } from '@common/utils/Validation'
+
+import { EnvironmentList } from './EnvironmentList'
+import EditEnvironmentGroupModal from './EditEnvironmentGroupModal'
+import { PageHeaderTitle, PageHeaderToolbar } from './EnvironmentGroupDetailsPageHeader'
+import { cleanData, EnvironmentGroupDetailsTab } from '../utils'
+
+import css from './EnvironmentGroupDetails.module.scss'
+
+const yamlBuilderReadOnlyModeProps: YamlBuilderProps = {
+  fileName: `environmentGroup.yaml`,
+  entityType: 'EnvironmentGroup',
+  width: '100%',
+  height: 300,
+  showSnippetSection: false,
+  yamlSanityConfig: {
+    removeEmptyString: false,
+    removeEmptyObject: false,
+    removeEmptyArray: false
+  }
+}
+
+export default function EnvironmentGroupDetails() {
+  const { accountId, orgIdentifier, projectIdentifier, environmentGroupIdentifier } = useParams<
+    ProjectPathProps & EnvironmentGroupPathProps
+  >()
+  const { sectionId } = useQueryParams<EnvironmentGroupQueryParams>()
+  const { updateQueryParams } = useUpdateQueryParams<EnvironmentGroupQueryParams>()
+
+  const { getString } = useStrings()
+  const { showSuccess, showError, clear } = useToaster()
+
+  const formikRef = useRef<FormikProps<EnvironmentGroupResponseDTO>>()
+
+  const [selectedTabId, setSelectedTabId] = useState<EnvironmentGroupDetailsTab>(
+    EnvironmentGroupDetailsTab[EnvironmentGroupDetailsTab[defaultTo(sectionId, 'CONFIGURATION')]]
+  )
+  const [selectedEnvs, setSelectedEnvs] = useState<EnvironmentResponse[]>([])
+  const [yamlHandler, setYamlHandler] = useState<YamlBuilderHandlerBinding | undefined>()
+  const [selectedView, setSelectedView] = useState<SelectedView>(SelectedView.VISUAL)
+  const [updateLoading, setUpdateLoading] = useState(false)
+  const [firstLoad, setFirstLoad] = useState(true)
+
+  const { data, loading, error, refetch } = useGetEnvironmentGroup({
+    queryParams: {
+      accountIdentifier: accountId,
+      orgIdentifier,
+      projectIdentifier
+    },
+    envGroupIdentifier: defaultTo(environmentGroupIdentifier, '')
+  })
+
+  useEffect(() => {
+    if (!loading && firstLoad) {
+      setFirstLoad(false)
+    }
+  }, [loading])
```

Comment:
Codacy has a fix for the issue: [React Hook useEffect has a missing dependency: 'firstLoad'. Either include it or remove the dependency array.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9264786)
```suggestion
  }, [firstLoad, loading])
```